ion-schema/src/isl/isl_constraint.rs

```diff
@@ -490,7 +498,14 @@ pub(crate) enum IslConstraintImpl {
     // For ISL 1.0 which doesn't support `distinct` elements this will be (type_reference, false).
     Element(IslTypeRefImpl, bool),
     Exponent(Range),
-    Fields(HashMap<String, IslTypeRefImpl>),
+    // Represents Fields(fields, content_closed)
+    // For ISL 2.0 true/false is specified based on whether `closed::` annotation is present or not
+    // For ISL 1.0 this will always be (fields, false) as it doesn't support `closed::` annotation on fields constraint
+    Fields(HashMap<String, IslTypeRefImpl>, bool),
```

Comment:
Does not need to be addressed in this PR. However, do you think we should change this to `Fields(Fields)` and add a `Fields { fields: HashMap<...>, is_closed: bool }` struct? (And similarly for the other constraints?)

@zslayton, do you have an opinion too?

Reply:
We can also do something like following if the intention is to have a name suggesting what the enum vvariant arguments are:
```rust
enum IslConstraintImpl {
   ...
   Fields({ field: HashMap<...>, is_closed: bool })
   ...
}
```

Reply:
If we're doing that, we may as well get rid of the tuple.
```rust
enum IslConstraintImpl {
   ...
   Fields { field: HashMap<...>, is_closed: bool }
   ...
}
```

What I'm really getting at, I suppose, is whether the enum variants should hold an opaque type.

Reply:
Sorry. I didn't mean to include the tuple in my example. I understand what you are suggesting here but I don't see it to be necessary since this enum is crate level enum now. I will create an issue to keep this discussion open though.

Reply:
Oh, I thought that since this was in `isl_constraint.rs` that this was the public version of the model. Is it not?

If it's not a public enum, then I have no objection to keeping it the way it is.

Reply:
Yes, its in `isl_constraint` but now I have added another struct `IslConstraint` which is the on visible publicly and this enum is just used by `IslCosntraint`.